### ISSUE TITLE
how to directly input my own low-resolution image (without high-resolution downsampling)

### ISSUE THREAD

**Issue by zymize** [xinntao/BasicSR#260]
Sorry to disturb you.I have one question.

How to test my own pictures using this code in tensorflow or pytorch? I want to restore my low resolution image to reconstruction ×2 resolution. In my opinion, the test is always to downsample the high-resolution test set to 2 times, and then reconstruct it. I don't know how to directly input my own low-resolution image (without high-resolution downsampling),and then reconstruct it to twice its original resolution?

Looking forward to your reply. Thank you very much.

**Comment by xinntao** [comment]
Hi @zymize , sorry for the late reply. 

BasicSR supports upsampling with only LR inputs. 

You can have a look at this option file. [options/test/SRResNet_SRGAN/test_MSRResNet_x4_woGT.yml](https://github.com/xinntao/BasicSR/blob/master/options/test/SRResNet_SRGAN/test_MSRResNet_x4_woGT.yml)

It uses the [SingleImageDataset](https://github.com/xinntao/BasicSR/blob/master/basicsr/data/single_image_dataset.py) ~
